diff --git a/Dockerfile b/Dockerfile
index ba908af..3a870d1 100644
--- ./Dockerfile
+++ ./Dockerfile
@@ -1,32 +1,10 @@
 FROM archlinux:base-devel
 
 ARG USERNAME
+WORKDIR /root
 
-RUN	pacman --noconfirm -Syu \
-		fish \
-		git \
-		meson \
-		neovim \
-		ninja \
-	&& pacman --noconfirm -Scc \
-	# make neovim available to use as vim
-	# below is cited from https://aur.archlinux.org/packages/neovim-symlinks
-	&& cd /usr/bin \
-	&& echo -e '#!/bin/sh\nexec nvim -e "$@"'  > ex \
-	&& echo -e '#!/bin/sh\nexec nvim -E "$@"'  > exim \
-	&& echo -e '#!/bin/sh\nexec nvim -RZ "$@"' > rview \
-	&& echo -e '#!/bin/sh\nexec nvim -Z "$@"'  > rvim \
-	&& echo -e '#!/bin/sh\nexec nvim -R "$@"'  > view \
-	&& echo -e '#!/bin/sh\nexec nvim -d "$@"'  > vimdiff \
-	&& chmod 755 ex* rvi* view vimdiff \
-	&& for _link in edit vedit vi vim; do ln -s nvim $_link; done
+COPY ./*.sh /root/
+RUN /root/setup.sh 
 
 COPY pacman.conf /etc/pacman.conf
 COPY sudoers /etc/sudoers
-
-RUN useradd -m -G wheel "$USERNAME"
-
-USER "$USERNAME"
-WORKDIR "/home/$USERNAME/"
-
-CMD /usr/bin/fish
diff --git a/docker-compose.yaml b/docker-compose.yml
similarity index 100%
rename from docker-compose.yaml
rename to docker-compose.yml
diff --git a/setup.sh b/setup.sh
new file mode 100755
index 0000000..ae1e413
--- /dev/null
+++ ./setup.sh
@@ -0,0 +1,8 @@
+#!/bin/bash
+pacman --noconfirm -Syu \
+    fish \
+    git \
+    meson \
+    neovim \
+    ninja \
+    && pacman --noconfirm -Scc
